windows/tools/compile_physics.py

```diff
@@ -1,75 +1,72 @@
 import os
 
+from pathlib import Path
 from traceback import format_exc
 
 from reclaimer.model.jms import read_jms
 from reclaimer.physics.physics_compilation import compile_physics
-
+from supyr_struct.util import is_path_empty
 from binilla.windows.filedialog import askopenfilename
 
+
 def physics_from_jms(app, fp=None):
     load_dir = app.jms_load_dir
     tags_dir = app.tags_dir
     data_dir = app.data_dir
-    if not tags_dir:
-        tags_dir = ""
-    if not data_dir:
-        data_dir = ""
+    if is_path_empty(tags_dir):
+        tags_dir = Path("")
+    if is_path_empty(data_dir):
+        data_dir = Path("")
 
-    if not load_dir:
+    if is_path_empty(load_dir):
         load_dir = data_dir
 
-    if not fp:
+    if is_path_empty(fp):
         fp = askopenfilename(
             initialdir=load_dir, parent=app,
             filetypes=(("JMS model", "*.jms"), ("All", "*")),
             title="Select jms file to turn into a physics tag")
 
-    if not fp:
+    fp = Path(fp)
+    if is_path_empty(fp):
         return
 
     try:
-        app.jms_load_dir = os.path.dirname(fp)
-
-        print("Creating physics from this jms file:")
-        print("    %s" % fp)
-        with open(fp, "r") as f:
+        app.jms_load_dir = fp.parent
+        with fp.open("r") as f:
             jms_model = read_jms(f.read(), "regions")
     except Exception:
         print(format_exc())
         print("    Could not parse jms file")
         return
+    
+    try:
+        rel_filepath = fp.relative_to(data_dir).parent.parent
+        rel_filepath = rel_filepath.joinpath("%s.physics" % rel_filepath.stem)

```

Comment:
.with_suffix()

Reply:
with_suffix raises an exception if the suffix isn't empty and doesn't start with a period. Unless I'm missing your point.

Reply:
Suffix has no crash condition for when there already is a suffix.
https://docs.python.org/3.5/library/pathlib.html#pathlib.PurePath.with_suffix
```suggestion
        rel_filepath = rel_filepath.with_suffix(".physics")
```
Also, what you wanted to do you needed the name for.
https://docs.python.org/3.5/library/pathlib.html#pathlib.PurePath.name

This code right now would turn `this/lovely/long/path.ext` into `this/lovely/long/path/this/lovely/long/path.physics`